### PR TITLE
fix(backup restore): fix backup volume synced delay restore failed

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#6213](https://github.com/longhorn/longhorn/issues/6213)

### CODE REVIEW

File: webhook/resources/volume/mutator.go [hunk 1]
```diff
@@ -143,7 +143,18 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
...
+		backup, err := v.ds.GetBackupRO(bName)
+		if err != nil {
+			return nil, werror.NewInvalidError(fmt.Sprintf("cannot get backup %s: %v", bName, err), "")
+		}
+
+		if bv != nil && backup != nil && backup.Status.VolumeBackingImageName != "" {
+			if bv.Status.BackingImageName != backup.Status.VolumeBackingImageName {
+				err = fmt.Errorf("backup volume's backing image %v does not match backup's backing image %v", bv.Status.BackingImageName, backup.Status.VolumeBackingImageName)
+				return nil, werror.NewInvalidError(err.Error(), "")
+			}
```
Comment (derekbit) [inline webhook/resources/volume/mutator.go:156]:
```suggestion
			if bv.Status.BackingImageName != backup.Status.VolumeBackingImageName {
				err = fmt.Errorf("backup volume's backing image %v does not match backup's backing image %v and please retry later.", bv.Status.BackingImageName, backup.Status.VolumeBackingImageName)
				return nil, werror.NewForbiddenError(err. Error())
			}
```

**Review by innobead — DISMISSED** [review]
LGTM, but please adopt Derek's suggestion for the error log.

**Comment by innobead** [comment]
@mergify backport v1.5.x

File: webhook/resources/volume/mutator.go (hunk 1 above)
Comment (ChanYiLin) [reply webhook/resources/volume/mutator.go]: fixed, thanks!

**Comment by ChanYiLin** [comment]
@innobead Fixed, thanks!

**Review by innobead — APPROVED** [review]

**Comment by ChanYiLin** [comment]
@mergify backport v1.4.x

**Comment by ChanYiLin** [comment]
@mergify backport v1.3.x

**Pull request merged**